src/dotnet/commands/dotnet-restore/Program.cs

```
@@ -131,11 +131,12 @@ private static void RestoreTool(LibraryRange tooldep, RestoreTask restoreTask, b
             {
                 var tempPath = Path.Combine(tempRoot, Guid.NewGuid().ToString(), "bin");
 
-                RestoreToolToPath(tooldep, restoreTask.Arguments, tempPath, quiet);
-
-                CreateDepsInPackageCache(tooldep, tempPath);
-
-                PersistLockFile(tooldep, tempPath, restoreTask.ProjectDirectory);
+                var restoreSucceded = RestoreToolToPath(tooldep, restoreTask.Arguments, tempPath, quiet);
+                if (restoreSucceded)
+                {
```

How can we get some diagnostics here? This is a silent failure. NRE was nicer since it indicates a problem...